File: frontend/lib/data/models/relative_search_response.dart
class ProfileSearchResponse {
  final String id;
  final String fullName;
  final String phoneNumber;
  final String dateOfBirth;
  final String avatarUrl;
  final String relationship;
  final String identityNumber;
  final String address;

  ProfileSearchResponse({
    required this.id,
    required this.fullName,
    required this.phoneNumber,
    required this.dateOfBirth,
    required this.avatarUrl,
    required this.relationship,
    required this.identityNumber,
    required this.address,
  });

  factory ProfileSearchResponse.fromJson(Map<String, dynamic> json) {
    return ProfileSearchResponse(
      id: json['id'] ?? '',
      fullName: json['fullName'] ?? 'Unknown List',
      phoneNumber: json['phoneNumber'] ?? '',
      dateOfBirth: json['dateOfBirth'] ?? '',
      avatarUrl: json['avatarUrl'] ?? '',
      relationship: json['relationship'] ?? '',
      identityNumber: json['identityNumber'] ?? '',
      address: json['address'] ?? '',
    );
  }
}

File: backend/src/main/java/backend/model/dto/response/RelativeSearchResponse.java
package backend.model.dto.response;

import lombok.Builder;
import lombok.Value;
import java.util.UUID;

@Value
@Builder
public class RelativeSearchResponse {
    UUID id;
    String fullName;
    String phoneNumber;
    String dateOfBirth;
    String avatarUrl;
    String relationship;
    String identityNumber;
    String address;
}
